File: app/src/main/java/com/wojdor/hearthstonecards/data/service/CardService.java
package com.wojdor.hearthstonecards.data.service;

public final class CardService {

    private static final String BASE_URL = "https://omgvamp-hearthstone-v1.p.mashape.com/";
    private static final Object LOCK = new Object();

    private static CardApi instance;

    private CardService() {
    }

    public static CardApi getInstance() {
        if (instance == null) {
            synchronized (LOCK) {
                instance = ServiceGenerator.getInstance(BASE_URL).createService(CardApi.class);
            }
        }
        return instance;
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/data/service/CardService.kt
package com.wojdor.hearthstonecards.data.service

class CardService {

    val cardApi by lazy {
        ServiceGenerator(BASE_URL).createService(CardApi::class.java)
    }

    companion object {
        private const val BASE_URL = "https://omgvamp-hearthstone-v1.p.mashape.com/"
    }
}
